### PR TITLE
Bower publishing

### PR DESCRIPTION
**Pull request by joeljuca**
**You must publish your package** on Bower (and eventually on npm, RubyGems and any other sort of package manager that people use to manage front-end packages).

Putting it simple: if you don't, people who want to use it will just download and inject it into their project's source code repository. Monkey patches will often happen because they will not create a patch and PR it, because it will be _"just a script they'd downloaded from the Internet."_ You also loose the ability to provide improvements to your users, etc.

So, publishing it to package managers is a must. So, I'm starting the Bower publishing here because that's what I need now (my case: Ruby on Rails app, using RubyGems and proxying some front-end assets through ruby-assets.org), but npm probably should be the next one.

So, they are still tasks for you:
- [x] **Bootstrap a `bower.json`**
- [ ] **Choose a name**  
  `jquery-loading-overlay` is already being used
- [ ] **Double-check the jQuery version dependency**  
  What is the oldest jQuery version your library is compatible with (or you want it to be)?
- [ ] **Are these keywords OK/enough?**
- [ ] **Publish it to the registry (bower.io)**  
  Follow steps from the [Register section](https://bower.io/docs/creating-packages/#register) from Bower's documentation

These are things you must do by your own, since you're the author (and now the maintainer) of the lib, it must be on your  package registries' realm.

### CODE REVIEW

**Comment by joeljuca** [comment]
FYI: RubyGems is probably not needed, since there's ruby-assets.org, a tool that makes Bower packages available for Ruby/Rails projects, and there's no need to process it somewhere before serving to the browser.

**Comment by gasparesganga** [comment]
I apprecciate the _emphasis_ you put in your post, really.
A pretty simple plugin is becoming quite popular and more and more users are giving me their feedback and I'm glad for that.
Since a few users are requesting bower or npn (or composer, for PHP projects) repositories and they feel a CDN is not enough, I am going to release my plugins on those platform.

Still, I need to be in _full control_ of my repositories because I'm releasing them in my spare time and most of them comes from some other projects I'm working on. In order to mantain the level of quality in all the code I release and mantain, I (unfortunately) **MUST** follow a strict route of _writing-testing-releasing_.

Just to be clear, this is what wrote in a post about a PHP library of mine ([PHP Shapefile](https://github.com/gasparesganga/php-shapefile)):

> my approach is to _basically never_ accept pull requests, I prefer to read suggestions and write/review all the code myself. My open source projects are actually forks of some proprietary libraries I write and mantain for my clients, thus the development and deployment flow goes **from them to these public releases**.

 I want to plan this accurately before diving into it. For my personal likes, I think a CDN is more appropriate for a jQuery plugin, but since I can easily make everybody happy, why not publishing on bower and npm too?

In conclusion, I am rejecting your pull request because of the reasons above, but please be patient for a few days and you will be able to find all my plugins in bower and/or npm too ;)

**Comment by joeljuca** [comment]
I don't understand what's the point of never accepting a pull request to one of your libraries. You are losing the biggest benefit of the Open Source. The pull request allows you to read, review and test the code before accepting it so there's no point here. And, if it comes from one of your clients' proprietary software you can even be breaking a contract by releasing that code as MIT licensed software.

So, I think you clearly don't understand the how and why of the open source. I'm writing this answer here as my last attempt to make you understand how pointless is your attitude of rejecting contributions from the community. You clearly are losing an opportunity to gain all the benefits of being a friendly and receptive community member of the Webdev community.

I hope you change your mind, there's still time for it.

**Comment by joeljuca** [comment]
And about package publishing, some projects just don't use CDN. My case: we work with heavily optimized bundles, so every vendor asset is concatenated, minified, compressed and tagged with a hash to maximize the utilization of browser caches, even through multiple deployments. All vendors with just one HTTP request (the real expensive thing those days).

We don't need/want a CDN, so we must have the packages available from package managers.

**Comment by gasparesganga** [comment]
Hey mate, I tried to be polite and all, but please don't try to _impose_ your ideas over the others'.

I will be polite again and answer both of your points, but at the end of the day I am releasing my libraries as **free** and  **open source** projects for everybody to benefit from, if they like. No one is obliged to use them, neither I am in no way obliged to change a single comma if I don't want to.

### Open Source

_Open source_ does **NOT** mean that you should accept code and reviews from other people. That's one of the uses, but not the one and only point of it. You will be surprised to know how many companies release their source code but keep the development as an internal business. Most of them even charge the users for the software. The point for them releasing the source code is to be as clear as possible about their product and show their users the quality of their work.
By the way, no, I am not breaking any contract releasing my libraries. The relationship between my clients and me and what I can and cannot do with the code I write, is absolutely up to **our** agreements, and of course I can release under the MIT license my libraries. I've chosen this kind of license especially for that: users who don't agree with my developing and releasing policies can fork the projects and work on their own version.
Again, there are many uses and licenses for open source. _Collaboration_ is one of them, but definitely not the only one.

### CDN

You said it right: _in your **case** you don't **need/want** a CDN_. That's fine. But many other developers prefer to have single libraries deployed through CDNs and eventually rely on their users browsers' caches built in a _cross-site_ fashion. In their opinion the pitfall of downloading a minimized file from a CDN overplays the greater benefit of not dowloading anything at all if the user already has that specific resource cached even before having ever visited their website. Leaving apart the complexity added by a custom and heavily optimized environment like the one you are using.
That's a matter of preferences and specific scenarios (and there are plenty of both!). And, believe me, through the years and experience I have learned that the more one thinks his view is _the only right way_, the more other people will eventually _prove him wrong_ providing a completely different set up.

### In conclusion

Anyways, I will open some kind of _info repository_ to better explain **my** reasons not to accept any pull requests (which I more or less have already summarized), so that people will be able to read them and understand.
And, as I have already written, I will pubblish my jQuery plugins on bower and/or npm, so people with different requirements and preferences will be able to benefit from them.
Finally, I am glad if people like my libraries and use them. And I apprecciate all kind of feedback and ideas they propose.

Cheers.

**Comment by gasparesganga** [comment]
Bower and npm packages available since v1.5.1, see [here](https://github.com/gasparesganga/jquery-loading-overlay/issues/7#issuecomment-260017029)
